lib/puppet/util.rb

```diff
@@ -25,6 +25,67 @@ module Util
 
   extend Puppet::Util::SymbolicFileMode
 
+  public
+  def self.get_environment_variable(name, mode = nil)
+    if mode.nil?
+      if Puppet.features.microsoft_windows?
+        mode = :windows
+      else
+        mode = :posix
+      end
+    end
+
+    if mode == :windows
+      Puppet::Util::Windows::Process.get_environment_strings.each do |key, value |
+        if name.casecmp(key) == 0 then
+          return value
+        end
+      end
+      return nil
+    else
+      ENV[name]
+    end
+  end
+  # TODO use alias_method instead
+  def self.env(name,mode = nil)
```

Comment:
So the big question is, do we 'patch' ENV or implement a new module called Puppet::Env or Puppet::Util::Env?

We can't use Puppet::Environments as that's for puppet environment